lib/rules/redux-actions-return-objects.js

```diff
@@ -0,0 +1,40 @@
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prohibits returning anything but objects from Redux actions',
+    },
+    fixable: false,
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          whiteList: {
+            default: [],
+            type: 'array',
+          },
+        },
+      },
+    ],
+  },
+
+  create(context) {
+    return {
+      'ReturnStatement'(node) {
+        const filename = context.getFilename();
+        const configuration = context.options[0] || {};
+        const whitelist = configuration.whitelist || [];
+
+        const inActions = filename.includes('state/actions/');
+        if (!inActions) { return; }
+
+        const returnsObject = node.argument && node.argument.type === 'ObjectExpression';
+        if (returnsObject) { return; }
```

Comment:
The description of this rule states that an `Action` must return an `Object`, but the way this is written, a raw `return;` statement would be valid as well, allowing `null` or `undefined` to get through.

Could this statement be removed in favor of changing the selector to `'ReturnStatement[argument.type!=ObjectExpression]'`?

Reply:
I think you maybe misread it - if you do `return;` then node.argument isn't defined, so `returnsObject` is falsey. There is a test that covers this very case.

That said, changing the selector seems good to me.

Reply:
Ah, my bad.